app/controllers/users/reset_passwords_controller.rb

```diff
@@ -91,6 +85,19 @@ def handle_valid_email
       redirect_to forgot_password_url(resend: resend_confirmation)
     end
 
+    def validate_password_reset_token
+      return if session[:reset_password_token]
+      session[:reset_password_token] = params[:reset_password_token]
```

Comment:
Should we delete the session value at some point, e.g. after the user changes their password?